apps/clamav.sh

```diff
@@ -0,0 +1,186 @@
+sed -i '/\[Service\]/a RestartSec=3' /etc/systemd/system/clamav-daemon.service
+check_command systemctl daemon-reload
+check_command systemctl restart clamav-daemon
+
+# Install Nextcloud app
+install_and_enable_app files_antivirus
+
+# Configure Nextcloud app
+occ_command config:app:set files_antivirus av_mode --value="socket"
+occ_command config:app:set files_antivirus av_socket --value="/var/run/clamav/clamd.ctl"
+occ_command config:app:set files_antivirus av_stream_max_length --value="104857600"
+occ_command config:app:set files_antivirus av_max_file_size --value="-1"
+occ_command config:app:set files_antivirus av_infected_action --value="only_log"
+
+# Inform the user
+msg_box "ClamAV was succesfully installed.
+
+Your Nextcloud should be more secure now."
+
+# Ask for full-scan
+if ! yesno_box_yes "Do you want to setup a weekly full scan of all your files?
+It will run on Sundays starting at 10:00 and will continue for a maximum of 12 hours (hardcoded). 
+You will be notified when it's finished so that you can check the final result."
+then
+    exit
+fi
+
+choice=$(whiptail --title "$TITLE" --nocancel --menu \
+"Choose what should happen with infected files.
+Infected files will always get reported to you no matter which option you choose.
+$MENU_GUIDE\n\n$RUN_LATER_GUIDE" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+"Only log" "" \
+"Copy to a folder" "" \
+"Move to a folder" "" \
+"Remove" "" 3>&1 1>&2 2>&3)
+
+case "$choice" in
+    "Only log")
+        ARGUMENT=""
+        AV_PATH=""
+    ;;
+    "Copy to a folder")
+        ARGUMENT="--move="
+        AV_PATH="/root/.clamav/clamav-fullscan.jail"
+        msg_box "We will copy the files to '$AV_PATH'"
+        mkdir -p "$AV_PATH"
+        chown -R clamav:clamav "$AV_PATH"
+        chmod -R 600 "$AV_PATH"
+    ;;
+    "Move to a folder")
+        ARGUMENT="--copy="
+        AV_PATH="/root/.clamav/clamav-fullscan.jail"
+        msg_box "We will move the files to '$AV_PATH'"
+        mkdir -p "$AV_PATH"
+        chown -R clamav:clamav "$AV_PATH"
+        chmod -R 600 "$AV_PATH"
+    ;;
+    "Remove")
+        ARGUMENT="--remove=yes"
+        AV_PATH=""
+    ;;
+    "")
+        exit 1
+    ;;
+    *)
+    ;;
+esac
+
+# Create the full-scan script
+cat << CLAMAV_REPORT > "$SCRIPTS"/clamav-fullscan.sh
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+AV_REPORT="$(clamscan \
+--recursive \
+--stdout \
+--infected \
+--cross-fs \
+--log=$VMLOGS/clamav-fullscan.log \
+$ARGUMENT$AV_PATH \
+--max-scantime=43200000 \
+--max-filesize=1G \
+--pcre-max-filesize=1G \
+--max-dir-recursion=30 \
+/ )"
+
+notify_admin_gui \
+"Your weekly full-scan ClamAV report" \
+"$AV_REPORT"
+CLAMAV_REPORT
+
+# Make the script executable
+chmod +x "$SCRIPTS"/clamav-fullscan.sh
+
+# Create the cronjob
+crontab -u root -l | { cat; echo "0 10 * * 7 $SCRIPTS/clamav-fullscan.sh 2>&1"; } | crontab -u root -
```

Comment:
Have you been taking the automatic updates script in consideration here? I'm thinking of the reboots....

Reply:
I thought updates happen on saturday evening?

Reply:
https://github.com/nextcloud/vm/blob/476ea4e96a65dd37a0b0da9a9335c793d6fd33ee/addons/automatic_updates.sh#L39

Reply:
OK great!